src/osversion/VersionTranslator.cpp

```diff
@@ -7664,6 +7945,348 @@ namespace osversion {
+              // * Maximum Outdoor Dry-Bulb Temperature for Crankcase Heater Operation - 14 => 18
+              newObject.setString(i + 4, value.get());
+            }
+          }
+        }
+
+        auto it = CoilLatentTransitionInfo::findFromCoolingCoil(coilTransitionInfos, object);
+        const bool hasCoilInfo = (it != coilTransitionInfos.end());
+        // No PLF Curve needed here
+
+        double maxCyclingRate = 2.5;
+        double heatPumpTimeConst = 60.0;
+        double hpDelayTime = 60.0;
+        if (hasCoilInfo) {
+          maxCyclingRate = it->maxCyclingRate;
+          heatPumpTimeConst = it->heatPumpTimeConst;
+          hpDelayTime = it->hpDelayTime;
+        }
+
+        newObject.setDouble(9, maxCyclingRate);
+        newObject.setDouble(10, heatPumpTimeConst);
+        newObject.setDouble(11, hpDelayTime);
+
+        // Cranckcase curve is optional
+        // newObject.setString(17; "");
+
+        m_refactored.push_back(RefactoredObjectData(object, newObject));
+        ss << newObject;
+
+      } else if (iddname == "OS:Coil:Cooling:WaterToAirHeatPump:EquationFit") {
+
+        // Fields that have been added from 3.6.1 to 3.7.0:
+        // ------------------------------------------------
+        // * Part Load Fraction Correlation Curve Name* 17
+        // New at end
+        // * Maximum Cycling Rate * 20
+        // * Latent Capacity Time Constant * 21
+        // * Fan Delay Time * 22
+
+        auto iddObject = idd_3_7_0.getObject(iddname);
+        IdfObject newObject(iddObject.get());
+
+        for (size_t i = 0; i < object.numFields(); ++i) {
+          if ((value = object.getString(i))) {
+            if (i < 17) {
+              newObject.setString(i, value.get());
+            } else {
+              newObject.setString(i + 1, value.get());
+            }
+          }
+        }
+
+        auto it = CoilLatentTransitionInfo::findFromCoolingCoil(coilTransitionInfos, object);
+        const bool hasCoilInfo = (it != coilTransitionInfos.end());
+        const std::string curveName = hasCoilInfo ? it->curveName() : fmt::format("{}-PLFCorrelationCurve", object.nameString());
+        newObject.setString(17, curveName);
+
+        double maxCyclingRate = 0.0;
+        double heatPumpTimeConst = 0.0;
+        double hpDelayTime = 60.0;
+        if (hasCoilInfo) {
+          maxCyclingRate = it->maxCyclingRate;
+          heatPumpTimeConst = it->heatPumpTimeConst;
+          hpDelayTime = it->hpDelayTime;
+        }
+
+        newObject.setDouble(20, maxCyclingRate);
+        newObject.setDouble(21, heatPumpTimeConst);
+        newObject.setDouble(22, hpDelayTime);
+
+        m_refactored.push_back(RefactoredObjectData(object, newObject));
+        ss << newObject;
+
+        if (!hasCoilInfo) {
+          IdfObject plfCurve = CoilLatentTransitionInfo::defaultHeatPumpCoilPLFCorrelationCurve(idd_3_7_0, curveName);
+          ss << plfCurve;
+          m_new.emplace_back(std::move(plfCurve));
+        }
+
```

Comment:
When you get to a coil that is supposed to inherit from a **potential** unitary parent

If you have a unitary parent:
* You use the values from the unitary, and the PLF curve name we generated automatically

You don't have a unitary parent:

* You use the defaults from the IDD (moved to Ctor in model since I made it required-field)
* You create a PLF curve calculated from these defaults